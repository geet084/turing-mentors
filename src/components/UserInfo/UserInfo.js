import React, { Component } from 'react';
import { CreateInput } from '../CreateInput/CreateInput';
import { CreateCheckbox } from '../CreateCheckbox/CreateCheckbox';

export class UserInfo extends Component {
  state = {
    first_name: '',
    last_name: '',
    identities: [],
    cohort: 0,
    program: '',
    current_job: '',
  }

  submitForm = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userBio']);
  }

  updateIdentity = ({ target }) => {
    const state = this.state[target.name];
    const value = parseInt(target.value);
    let updatedState;

    if (state.includes(value)) updatedState = state.filter(prop => prop !== value)
    else updatedState = [...state, value]

    this.setState({ [target.name]: updatedState.sort() })
  }

  updateProgram = ({ target }) => {
    this.setState({ [target.name]: target.value })
  }

  updateCohort = ({ target }) => {
    this.setState({ [target.name]: parseInt(target.value) })
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value })
  }

  render() {
    const { first_name, last_name, cohort, current_job } = this.state;
    const mentor = this.props.user === 'Mentor';
    const identities = ['Male', 'Female', 'Non-Binary'].map((identity, i) => (
      <CreateCheckbox key={i} field="identities" name={identity} value={i + 1} checkBoxes={this.updateIdentity} />
    ))
    
    return (
      <div>
        <form onSubmit={this.submitForm} autoComplete='off'>
          <span className="pages">{this.props.user} User info:</span>
          <CreateInput field="first_name" text="First Name" value={first_name} handleChange={this.handleChange} max="28" />
          <CreateInput field="last_name" text="Last Name" value={last_name} handleChange={this.handleChange} max="28" />
          <div className="check-box">
            {identities}
          </div>
          <div>
            <CreateInput field="cohort" text="Cohort (ex: 1406)" value={cohort} handleChange={this.updateCohort} max='4' />
            <CreateCheckbox className="aa" field="program" name={'BE'} value={'BE'} checkBoxes={this.updateProgram} />
            <CreateCheckbox field="program" name={'FE'} value={'FE'} checkBoxes={this.updateProgram} />
          </div>
        </form>
        <div className="input-box">
          {
            mentor &&
            <CreateInput field="current_job" text="Current Job" value={current_job} handleChange={this.handleChange} max="28" />
          }
          <button className="next-btn" onClick={this.submitForm}>Next</button>
          <span className="pages">1 of {mentor ? '6' : '4'}</span>
        </div>
      </div>
    )
  }
}

export default UserInfo;