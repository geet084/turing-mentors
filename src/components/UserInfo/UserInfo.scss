.check-box {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  margin: 0px;
}

.pages {
  margin: 0 10px; 
  font-size: .9rem;
  color: #ccc;
}

.next-btn {
  color: #bbb;
  background-color: white;
  font-size: 1.25rem;
  margin: 0 10px;
  padding: 2px 4px;
  border: 1px solid white;
  border-radius: 10px;
  box-shadow: 4px 4px 8px -2px rgba(0,0,0,0.25);
  &:hover {
    cursor: pointer;
    margin-bottom: -2px;
    border-bottom: 2px solid #12f1fc;
  }
}