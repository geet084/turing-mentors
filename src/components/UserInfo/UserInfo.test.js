import React from 'react';
import { shallow } from 'enzyme';
import { UserInfo } from './UserInfo';

describe('UserInfo', () => {
  let wrapper;
  const mockProps = {
    updateUserInfo: jest.fn()
  };
  const mockState = {
    first_name: '',
    last_name: '',
    identities: [],
    cohort: 0,
    program: '',
    current_job: '',
  };

  beforeEach(() => {
    wrapper = shallow(<UserInfo {...mockProps} />);
  });

  describe('initial state & snapshot', () => {
    it('should match the correct snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have an default state', () => {
      const expected = mockState;

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('submitForm', () => {
    it('should handle the form submission', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'userBio'];

      wrapper.find('button').simulate('click', mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('updateIdentity', () => {
    it.skip('should update the selected identity in state', () => {
      const mockEvent = { target: { name: 'Female', value: 1 } };
      const expected = 'a';

      wrapper.find('.check-box').simulate('click', mockEvent);

      expect(wrapper.state()).toEqual(expected)

    });
  });

  describe('updateProgram', () => {
    it.skip('should update the program', () => {
      const mockEvent = { target: { name: 'BE', value: 'BE' } };
      const expected = 'a';

      wrapper.find('.aa').simulate('click', mockEvent);

      expect(wrapper.state()).toEqual(expected);
    });
  });
});