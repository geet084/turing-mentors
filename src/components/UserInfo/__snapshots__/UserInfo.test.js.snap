// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo initial state & snapshot should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserInfo
    updateUserInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <span
            className="pages"
          >
             User info:
          </span>
          <CreateInput
            field="first_name"
            handleChange={[Function]}
            max="28"
            text="First Name"
            value=""
          />
          <CreateInput
            field="last_name"
            handleChange={[Function]}
            max="28"
            text="Last Name"
            value=""
          />
          <div
            className="check-box"
          >
            <CreateCheckbox
              checkBoxes={[Function]}
              field="identities"
              name="Male"
              value={0}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="identities"
              name="Female"
              value={1}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="identities"
              name="Non-Binary"
              value={2}
            />
          </div>
          <div>
            <CreateInput
              field="cohort"
              handleChange={[Function]}
              max="4"
              text="Cohort (ex: 1406)"
              value={0}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              className="aa"
              field="program"
              name="BE"
              value="BE"
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="program"
              name="FE"
              value="FE"
            />
          </div>
        </form>,
        <div
          className="input-box"
        >
          <CreateInput
            field="current_job"
            handleChange={[Function]}
            max="28"
            text="Current Job"
            value=""
          />
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Next
          </button>
          <span
            className="pages"
          >
            1 of 
            4
          </span>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <span
              className="pages"
            >
               User info:
            </span>,
            <CreateInput
              field="first_name"
              handleChange={[Function]}
              max="28"
              text="First Name"
              value=""
            />,
            <CreateInput
              field="last_name"
              handleChange={[Function]}
              max="28"
              text="Last Name"
              value=""
            />,
            <div
              className="check-box"
            >
              <CreateCheckbox
                checkBoxes={[Function]}
                field="identities"
                name="Male"
                value={0}
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                field="identities"
                name="Female"
                value={1}
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                field="identities"
                name="Non-Binary"
                value={2}
              />
            </div>,
            <div>
              <CreateInput
                field="cohort"
                handleChange={[Function]}
                max="4"
                text="Cohort (ex: 1406)"
                value={0}
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                className="aa"
                field="program"
                name="BE"
                value="BE"
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                field="program"
                name="FE"
                value="FE"
              />
            </div>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " User info:",
              ],
              "className": "pages",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " User info:",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "first_name",
              "handleChange": [Function],
              "max": "28",
              "text": "First Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "last_name",
              "handleChange": [Function],
              "max": "28",
              "text": "Last Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="identities"
                  name="Male"
                  value={0}
                />,
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="identities"
                  name="Female"
                  value={1}
                />,
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="identities"
                  name="Non-Binary"
                  value={2}
                />,
              ],
              "className": "check-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "checkBoxes": [Function],
                  "field": "identities",
                  "name": "Male",
                  "value": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "checkBoxes": [Function],
                  "field": "identities",
                  "name": "Female",
                  "value": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "checkBoxes": [Function],
                  "field": "identities",
                  "name": "Non-Binary",
                  "value": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <CreateInput
                  field="cohort"
                  handleChange={[Function]}
                  max="4"
                  text="Cohort (ex: 1406)"
                  value={0}
                />,
                <CreateCheckbox
                  checkBoxes={[Function]}
                  className="aa"
                  field="program"
                  name="BE"
                  value="BE"
                />,
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="program"
                  name="FE"
                  value="FE"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "field": "cohort",
                  "handleChange": [Function],
                  "max": "4",
                  "text": "Cohort (ex: 1406)",
                  "value": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checkBoxes": [Function],
                  "className": "aa",
                  "field": "program",
                  "name": "BE",
                  "value": "BE",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "checkBoxes": [Function],
                  "field": "program",
                  "name": "FE",
                  "value": "FE",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CreateInput
              field="current_job"
              handleChange={[Function]}
              max="28"
              text="Current Job"
              value=""
            />,
            <button
              className="next-btn"
              onClick={[Function]}
            >
              Next
            </button>,
            <span
              className="pages"
            >
              1 of 
              4
            </span>,
          ],
          "className": "input-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "current_job",
              "handleChange": [Function],
              "max": "28",
              "text": "Current Job",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "next-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "1 of ",
                "4",
              ],
              "className": "pages",
            },
            "ref": null,
            "rendered": Array [
              "1 of ",
              "4",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <span
              className="pages"
            >
               User info:
            </span>
            <CreateInput
              field="first_name"
              handleChange={[Function]}
              max="28"
              text="First Name"
              value=""
            />
            <CreateInput
              field="last_name"
              handleChange={[Function]}
              max="28"
              text="Last Name"
              value=""
            />
            <div
              className="check-box"
            >
              <CreateCheckbox
                checkBoxes={[Function]}
                field="identities"
                name="Male"
                value={0}
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                field="identities"
                name="Female"
                value={1}
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                field="identities"
                name="Non-Binary"
                value={2}
              />
            </div>
            <div>
              <CreateInput
                field="cohort"
                handleChange={[Function]}
                max="4"
                text="Cohort (ex: 1406)"
                value={0}
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                className="aa"
                field="program"
                name="BE"
                value="BE"
              />
              <CreateCheckbox
                checkBoxes={[Function]}
                field="program"
                name="FE"
                value="FE"
              />
            </div>
          </form>,
          <div
            className="input-box"
          >
            <CreateInput
              field="current_job"
              handleChange={[Function]}
              max="28"
              text="Current Job"
              value=""
            />
            <button
              className="next-btn"
              onClick={[Function]}
            >
              Next
            </button>
            <span
              className="pages"
            >
              1 of 
              4
            </span>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "children": Array [
              <span
                className="pages"
              >
                 User info:
              </span>,
              <CreateInput
                field="first_name"
                handleChange={[Function]}
                max="28"
                text="First Name"
                value=""
              />,
              <CreateInput
                field="last_name"
                handleChange={[Function]}
                max="28"
                text="Last Name"
                value=""
              />,
              <div
                className="check-box"
              >
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="identities"
                  name="Male"
                  value={0}
                />
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="identities"
                  name="Female"
                  value={1}
                />
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="identities"
                  name="Non-Binary"
                  value={2}
                />
              </div>,
              <div>
                <CreateInput
                  field="cohort"
                  handleChange={[Function]}
                  max="4"
                  text="Cohort (ex: 1406)"
                  value={0}
                />
                <CreateCheckbox
                  checkBoxes={[Function]}
                  className="aa"
                  field="program"
                  name="BE"
                  value="BE"
                />
                <CreateCheckbox
                  checkBoxes={[Function]}
                  field="program"
                  name="FE"
                  value="FE"
                />
              </div>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " User info:",
                ],
                "className": "pages",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " User info:",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "first_name",
                "handleChange": [Function],
                "max": "28",
                "text": "First Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "last_name",
                "handleChange": [Function],
                "max": "28",
                "text": "Last Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <CreateCheckbox
                    checkBoxes={[Function]}
                    field="identities"
                    name="Male"
                    value={0}
                  />,
                  <CreateCheckbox
                    checkBoxes={[Function]}
                    field="identities"
                    name="Female"
                    value={1}
                  />,
                  <CreateCheckbox
                    checkBoxes={[Function]}
                    field="identities"
                    name="Non-Binary"
                    value={2}
                  />,
                ],
                "className": "check-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "class",
                  "props": Object {
                    "checkBoxes": [Function],
                    "field": "identities",
                    "name": "Male",
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "checkBoxes": [Function],
                    "field": "identities",
                    "name": "Female",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "checkBoxes": [Function],
                    "field": "identities",
                    "name": "Non-Binary",
                    "value": 2,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <CreateInput
                    field="cohort"
                    handleChange={[Function]}
                    max="4"
                    text="Cohort (ex: 1406)"
                    value={0}
                  />,
                  <CreateCheckbox
                    checkBoxes={[Function]}
                    className="aa"
                    field="program"
                    name="BE"
                    value="BE"
                  />,
                  <CreateCheckbox
                    checkBoxes={[Function]}
                    field="program"
                    name="FE"
                    value="FE"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "field": "cohort",
                    "handleChange": [Function],
                    "max": "4",
                    "text": "Cohort (ex: 1406)",
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checkBoxes": [Function],
                    "className": "aa",
                    "field": "program",
                    "name": "BE",
                    "value": "BE",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checkBoxes": [Function],
                    "field": "program",
                    "name": "FE",
                    "value": "FE",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CreateInput
                field="current_job"
                handleChange={[Function]}
                max="28"
                text="Current Job"
                value=""
              />,
              <button
                className="next-btn"
                onClick={[Function]}
              >
                Next
              </button>,
              <span
                className="pages"
              >
                1 of 
                4
              </span>,
            ],
            "className": "input-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "current_job",
                "handleChange": [Function],
                "max": "28",
                "text": "Current Job",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "next-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "1 of ",
                  "4",
                ],
                "className": "pages",
              },
              "ref": null,
              "rendered": Array [
                "1 of ",
                "4",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
